### ISSUE TITLE
Token auth and JWT term usage vs. JWS

### ISSUE THREAD

**Issue by unuseless** [miguelgrinberg/Flask-HTTPAuth#69]
In e.g. token_auth.py (and several other places) the term JWT is used for the tokens generated.  These tokens are generated with the itsdangerous TimedJSONWebSignatureSerializer.
The thing is, these are not JWT tokens, but JSON Web Signatures (JWS, see https://pythonhosted.org/itsdangerous/ ).

Remedy:  Don't use the term JWT for these tokens.

**Comment by miguelgrinberg** [comment]
Yes, you are absolutely right, I'll rename it.

**Comment by comatrion** [comment]
References remain e.g.
https://github.com/miguelgrinberg/Flask-HTTPAuth/blob/bd6fcf32dabec98a795dc6e3565f1ff9b66fcdfb/examples/multi_auth.py#L13
https://github.com/miguelgrinberg/Flask-HTTPAuth/blob/bd6fcf32dabec98a795dc6e3565f1ff9b66fcdfb/examples/token_auth.py#L11
https://github.com/miguelgrinberg/Flask-HTTPAuth/blob/bd6fcf32dabec98a795dc6e3565f1ff9b66fcdfb/docs/index.rst#L165
https://github.com/miguelgrinberg/Flask-HTTPAuth/blob/bd6fcf32dabec98a795dc6e3565f1ff9b66fcdfb/docs/index.rst#L170

**Comment by MisterY** [comment]
Does that mean that, if we were to use

```
def generate_token(self, user_id):
        """
        Generates the Auth Token
        :return: string
        """
        try:
            # set up a payload with an expiration time
            payload = {
                'exp': datetime.utcnow() + timedelta(minutes=5),
                'iat': datetime.utcnow(),
                'sub': user_id
            }
            # create the byte string token using the payload and the SECRET key
            secret = current_app.config.get('SECRET_KEY')
            jwt_string = jwt.encode(
                payload,
                secret,
                algorithm='HS256'
            )
            return jwt_string

        except Exception as e:
            # return an error in string format if an exception occurs
            return str(e)

    @staticmethod
    def decode_token(token):
        """
        Validates the auth token.
        Decodes the access token from the Authorization header.
        :param auth_token:
        :return: integer|string
        """
        try:
            # try to decode the token using our SECRET variable
            secret = current_app.config.get('SECRET_KEY')
            # is_blacklisted_token = BlacklistToken.check_blacklist(auth_token)
            payload = jwt.decode(token, secret)
            return payload['sub']
        except jwt.ExpiredSignatureError:
            # the token is expired, return an error string
            return "Expired token. Please login to get a new token"
        except jwt.InvalidTokenError:
            # the token is invalid, return an error string
            return "Invalid token. Please register or login"
```
the reference to JWT would be correct? Is this all that is necessary to correctly utilize JWT tokens?

**Comment by unuseless** [comment]
@MisterY If I correctly understand your code, you utilize pyjwt (though no import statement in your code).  If you use pyjwt it's easy to use JWT tokens.
@miguelgrinberg 's code just uses itsdangerous, so no extra external dependency needed.
In terms of token usage, when you pass such a token back and forth in your app, it should not matter if you use JWS or JWT.

**Comment by MisterY** [comment]
@unuseless, that's correct. There's `import jwt` statement at the top (ref: [here](https://gitlab.com/alensiljak/gnucash-portfolio-webapi/blob/master/gnucash_portfolio_webapi/models.py)).
Thanks for the clarifications.
I was mostly concerned about using some standard way of handling tokens as the clients will, most likely, be JavaScript from a Progressive Web Applications or perhaps other mobile apps. I have not made a final decision. Too many options available but none of them fits my goals of being cross-platform and easy to maintain.

**Comment by miguelgrinberg** [comment]
Addressed by #79. Closing.
